plugins/policy-recommendation/antrea_crd.py

```diff
@@ -0,0 +1,627 @@
+            value = getattr(self, attr)
+            if isinstance(value, list):
+                result[attr] = list(map(
+                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
+                    value
+                ))
+            elif hasattr(value, "to_dict"):
+                result[attr] = value.to_dict()
+            elif isinstance(value, dict):
+                result[attr] = dict(map(
+                    lambda item: (item[0], item[1].to_dict())
+                    if hasattr(item[1], "to_dict") else item,
+                    value.items()
+                ))
+            else:
+                result[attr] = value
+
+        return result
+
+class GroupStatus(object):
+    attribute_types = {
+        "conditions": "list[GroupCondition]"
+    }
+
+    def __init__(self, conditions=None):
+        self.conditions = conditions
+    
+    def to_dict(self):
+        """Returns the model properties as a dict"""
+        result = {}
+
+        for attr, _ in six.iteritems(self.attribute_types):
+            value = getattr(self, attr)
+            if isinstance(value, list):
+                result[attr] = list(map(
+                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
+                    value
+                ))
+            elif hasattr(value, "to_dict"):
+                result[attr] = value.to_dict()
+            elif isinstance(value, dict):
+                result[attr] = dict(map(
+                    lambda item: (item[0], item[1].to_dict())
+                    if hasattr(item[1], "to_dict") else item,
+                    value.items()
+                ))
+            else:
+                result[attr] = value
+
+        return result
+
+class GroupCondition(object):
+    attribute_types = {
+      "type": "string",
+      "status": "string",
+      "last_transition_time": "datetime",
+    }
+
+    def __init__(self, type=None, status=None, last_transition_time=None):
+        self.type = type
+        self.status = status
+        self.last_transition_time = last_transition_time
+    
+    def to_dict(self):
+        """Returns the model properties as a dict"""
+        result = {}
+
+        for attr, _ in six.iteritems(self.attribute_types):
+            value = getattr(self, attr)
+            if isinstance(value, list):
+                result[attr] = list(map(
+                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
+                    value
+                ))
+            elif hasattr(value, "to_dict"):
+                result[attr] = value.to_dict()
+            elif isinstance(value, dict):
+                result[attr] = dict(map(
+                    lambda item: (item[0], item[1].to_dict())
+                    if hasattr(item[1], "to_dict") else item,
+                    value.items()
+                ))
+            else:
+                result[attr] = value
+
+        return result
+
+class ClusterNetworkPolicy(object):
```

Comment:
Just a curiosity: it seems you have redefined many objects from the python k8s client because they're used by AntreaNetworkPolicies.

This is fine, but I don't understand why you could not import those from the K8S client library

Reply:
I think I only redefined necessary objects that are not included in the python k8s client or that are different from Antrea's definition. For example, I didn't redefine the `metadata`. But I might be wrong, please let me know if you find any objects that could be reused from the k8s client.

Reply:
Ok, so you redefined only those for which there are changes in Antrea network policy.... that's not a problem. I would have just considered inheriting and replacing attributes, but it's fine also in this way,